src/wrappers/themis/rust/src/secure_message.rs

```diff
@@ -215,7 +215,7 @@ impl SecureSign {
     /// Makes a new Secure Message using given secret key.
     ///
     /// Both ECDSA and RSA keys are supported.
-    pub fn new<S: Into<SecretKey>>(secret_key: S) -> Self {
+    pub fn new(secret_key: impl Into<SecretKey>) -> Self {
```

Comment:
I have a strong opinion that we should add one more breaking change into this PR 😇

In cryptography, "secret key" and "secret key cryptography" means "symmetric key". Single key for encryption and for decryption.

Asymmetric cryptography operates with keypairs, one of the is _private_ key another is public key.

It makes no sense from cryptographic/cryptographer perspective to use "secret" key in asymmetric process SecureMessage/SecureSign/SecureVerify.

I strongly suggest to rename "SecretKey" type to "PrivateKey" type everywhere in rust wrapper (including RSASecretKey, ECSecretKey, and other places of [`keys.rs`](https://github.com/cossacklabs/themis/blob/master/src/wrappers/themis/rust/src/keys.rs) and all tests).

It also will increase usability, because [official documentation](https://github.com/cossacklabs/themis/wiki/Secure-Message-cryptosystem) and all other wrappers operate with "private-public" terminology.

Do you think this change should be a part of this PR, or next one?

Reply:
I don’t mind and agree with this reasoning. Usability and consistency are important.

I’d prefer to make this change as a separate pull request. Then it will be a separate commit in git history and won’t raise questions like ‘Why the commit named ”Do not use fixed used context in Secure Cell (#358)” touches a whole bunch of seemingly unrelated files?’

```console
$ git grep -i secret *.rs | wc -l
276
$ git grep -i secret *.rs | awk 'BEGIN{FS=":"} {print $1}' | sort | uniq
docs/examples/rust/keygen.rs
docs/examples/rust/secure_compare.rs
docs/examples/rust/secure_message_client_encrypt.rs
docs/examples/rust/secure_message_client_verify.rs
src/wrappers/themis/rust/src/keygen.rs
src/wrappers/themis/rust/src/keys.rs
src/wrappers/themis/rust/src/secure_comparator.rs
src/wrappers/themis/rust/src/secure_message.rs
src/wrappers/themis/rust/src/secure_session.rs
tests/rust/keys.rs
tests/rust/secure_cell.rs
tests/rust/secure_comparator.rs
tests/rust/secure_message.rs
tests/rust/secure_session.rs
tools/rust/keygen_tool.rs
tools/rust/smessage_encryption.rs
```

(My initial reasons for using “secret-public” naming instead of traditional one were pretty shallow: I was mesmerized by the fact that these words have the same length thus enabling neatly aligned columns in monospaced source code. Plus, the other codebase I've been working on at that moment is using this naming; despite libsodium using “secret” for symmetric keys.)